specifyweb/businessrules/agent_rules.py

```diff
@@ -6,20 +6,41 @@
 def agent_delete_blocked_by_related_specifyuser(agent):
     try:
         Specifyuser.objects.get(agents=agent)
+        user = Specifyuser.objects.get(agents=agent)
     except Specifyuser.DoesNotExist:
         return
-    raise BusinessRuleException("agent cannot be deleted while associated with a specifyuser")
+    raise BusinessRuleException(
+        "agent cannot be deleted while associated with a specifyuser", 
+        {"table" : "Agent",
+         "type" : "DELETE_AGENT_USER",
+         "agentid" : agent.id, 
+         "specifyuserid": user.id})
 
 # Disabling this rule because system agents must be created separate from divisions
 # @orm_signal_handler('pre_save', 'Agent')
 # def agent_division_must_not_be_null(agent):
 #     if agent.division is None:
-#         raise BusinessRuleException("agent.division cannot be null")
+#         raise BusinessRuleException(
+#             "agent.division cannot be null",
+#             {"table" : "Agent",
+#              "type" : "NOT_NULL",
+#              "agentid" : agent.id})
 
 @orm_signal_handler('pre_save', 'Agent')
 def agent_types_other_and_group_do_not_have_addresses(agent):
     from specifyweb.specify.agent_types import agent_types
     if agent.agenttype is None:
-        raise BusinessRuleException("agenttype cannot be null")
-    if agent_types[agent.agenttype] in ('Other', 'Group'):
-        agent.addresses.all().delete()
+        raise BusinessRuleException(
+            "agenttype cannot be null", 
```

Comment:
Can you check if this is always enforced on the front-end? If so, we won't need special handling for this case on the front-end

Reply:
Yes, looks like this is already handled on the front-end!